### PR TITLE
Adds pytest-forked dependency to fix pytest memory accumulation issues.

### PR DESCRIPTION
**Pull request by monorimet**
This adds pytest-forked to requirements.txt -- to use this plugin, simply add `--forked` to your pytest command.
This way, each individual test case is run in its own subprocess, and error handling is seamlessly piped from the subprocess to pytest's reports.

@mariecwhite this should unblock us on the TF GPU enablement.

### CODE REVIEW

File: .github/workflows/test-models.yml [hunk 1]
```diff
@@ -100,9 +100,9 @@ jobs:
       if: matrix.suite == 'cpu'
       run: |
         cd $GITHUB_WORKSPACE
-        PYTHON=python${{ matrix.python-version }} BENCHMARK=1 IMPORTER=1 ./setup_venv.sh
```
Comment (dan-garvey) [inline .github/workflows/test-models.yml:103]: is this no longer required

File: .github/workflows/test-models.yml [hunk 2]
```diff
@@ -126,12 +126,12 @@ jobs:
         export DYLD_LIBRARY_PATH=/usr/local/lib/
         echo $PATH
         pip list | grep -E "torch|iree"
-        pytest -s --ci --ci_sha=${SHORT_SHA} --local_tank_cache="/Volumes/builder/anush/shark_cache" tank/test_models.py -k vulkan --update_tank
+        pytest --ci --ci_sha=${SHORT_SHA} --local_tank_cache="/Volumes/builder/anush/shark_cache" -k vulkan --update_tank
```
Comment (dan-garvey) [inline .github/workflows/test-models.yml:129]: why does this work without a target for pytest?

**Review by dan-garvey — APPROVED** [review]
check comments are ok then merge

File: .github/workflows/test-models.yml (hunk 2 above)
Comment (monorimet) [reply .github/workflows/test-models.yml:129]: tank/test_models.py is the only valid target from the base SHARK directory -- the other ones are ignored via SHARK/pytest.ini

**Pull request merged**